### ISSUE TITLE
Incorrect usage of `role="group"` in block list item

### ISSUE THREAD

**Issue by kevin940726** [WordPress/gutenberg#24508]
**Describe the bug**
Seems like we're using a `role="group"`in some `<li>` blocks, which is invalid as suggested in the [spec](https://www.w3.org/TR/wai-aria-1.1/#list).

> Lists contain children whose role is `listitem`, or elements whose role is `group` which in turn contains children whose role is `listitem`.

Here is a snapshot of our **Navigation Link** block

```html
<li
  aria-label="Block: Link"
  role="group"
  class="wp-block-navigation-link block-editor-block-list__block wp-block has-link wp-block"
  id="block-ee88e35b-c6a5-477b-8d10-0a226e8f37c4"
  data-block="ee88e35b-c6a5-477b-8d10-0a226e8f37c4"
  data-type="core/navigation-link"
  data-title="Link"
  tabindex="0"
></li>
```

This is also tested and reported by the e2e Axe plugin.

**To reproduce**

1. Create or edit a post
2. Create a *Navigation (horizontal)* block
3. Add a Link
4. Add another Link in the submenu of the Link added in (3.)
5. Observe the HTML code of the submenu Link

**Expected behavior**
The `<li>` should not have `role="group"`, or we should have the children of the `<li>` have `role="listitem"`.

**Screenshots**
![image](https://user-images.githubusercontent.com/7753001/89980100-c6203180-dca3-11ea-9a25-862617a27283.png)

**Editor version (please complete the following information):**

- WordPress version: development version (5.6-alpha-48736)
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? default
- If the Gutenberg plugin is installed, which version is it? `master` branch

**Desktop (please complete the following information):**

- OS: macOS v10.15.6
- Browser: Chrome
- Version: 84.0.4147.105

**Comment by skorasaurus** [comment]
I'm not able to reproduce this with your instructions; the markup of the navigation block has changed quite a bit. 

an example with the markup from gutenberg 11 is below; ran it through axe with no problems. 

![image](https://user-images.githubusercontent.com/955351/129226898-adc609cd-6a52-423a-8fad-779630c13079.png)
